### PR TITLE
Update AbstractGuardAuthenticator implementation code to pass curl test according to documentation

### PR DESCRIPTION
**Pull request by DaumantasUrb**
Make the test curl http://localhost:8000/ return this value:
{"message":"Authentication Required"}
instead of
{"message":"Username could not be found."}

### CODE REVIEW

**Comment by xabbuh** [comment]
Hm, we recently made this change (see #7065).

**Comment by DaumantasUrb** [comment]
hmm sorry about that, first pull request, I can only see it being corrected in 3.1 and not the above versions

File: security/guard_authentication.rst [hunk 1]
```diff
@@ -172,8 +172,8 @@ This requires you to implement seven methods::
         public function getCredentials(Request $request)
         {
             if (!$token = $request->headers->get('X-AUTH-TOKEN')) {
-                // No token?
-                $token = null;
+                // No token? Return null and show "Authentication Required"
```
Comment (HeahDude) [inline security/guard_authentication.rst]: Hello @DaumantasUrb, thank you for this first contribution.
Sadly this change is wrong, another guard can still authenticate this request is this one returning `null`. If one want to get the exception at this point an exception should be thrown.

Should you we improve the docs so this is better explained?

**Comment by javiereguiluz** [comment]
I'm closing because we recently closed another similar pull request that also returned `null` instead of throwing the exception, which apparently is wrong in this case.

**Pull request closed without merging**